src/unix/getnameinfo.c

```diff
@@ -107,12 +158,31 @@ int uv_getnameinfo(uv_loop_t* loop,
   req->retcode = 0;
 
   if (getnameinfo_cb) {
+#if defined(_USE_LIBINFO)
+    int32_t get_nameinfo_result;
+    mach_port_t port;
+
+    get_nameinfo_result = getnameinfo_async_start(&port,
+                                                  (struct sockaddr*) &req->storage,
+                                                  salen,
+                                                  req->flags,
+                                                  uv__getnameinfo_async_done,
+                                                  req);
+    if (get_nameinfo_result == -1) {
+      return get_nameinfo_result;
+    }
```

Comment:
it makes no sense the thread pool to be used, if one fails the other will fail as well. Other than that, ok.

Reply:
How so? `getnameinfo_async_start()` does some initialization work that can fail but there's a decent chance regular getnameinfo() will still work.

Reply:
this is very unlikely.

OTOH handling both cases makes the code more complex, for instance `uv_cancel()` (ok, that one is like 2 lines of code more), but I'd rather go with a reduce code path and see if that comes back in practice (which I think will never come back).

Reply:
I guess that we can't escape from some kind of checking in `uv_cancel()`, since we're switching to `dlopen`/`dlsym` and we'll only know if we are using the async version at runtime. However I think that in this section, we should just fail in case the async version returns an error code.

Reply:
I would appreciate it if you and @barbieri motivate why you feel that way. My perspective is that falling through to regular getnameinfo() adds very little complexity.

Reply:
reducing code paths helps to avoid bugs due lack of testing of such code paths... but ok, agreed it's not a major thing, let's fallback :-)